### PR TITLE
Add metrics for time spent and timeouts in broker waiting for a connection.

### PR DESCRIPTION
**Pull request by rven1**
This metric will give an idea of the amount of time spent in the broker
waiting for a server connection to help us provision better.

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java [hunk 1]
```diff
@@ -32,6 +32,8 @@
   REQUEST_FETCH_EXCEPTIONS("exceptions", false),
   REQUEST_DESERIALIZATION_EXCEPTIONS("exceptions", false),
   DOCUMENTS_SCANNED("documents", false),
+  TIMEOUT_WAITING_FOR_CONNECTION("timeouts", false),
+  WAIT_FOR_CONNECTION("waits", false),
```
Comment (mayankshriv) [inline pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java]: Rename these to have REQUEST as part of the name so it is more readable.

File: pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java [hunk 2]
```diff
@@ -426,11 +426,13 @@ private void runInternal() {
       KeyedFuture<ServerInstance, NettyClientConnection> keyedFuture = null;
       boolean gotConnection = false;
       boolean error = true;
+      long timeRemaining = _timeoutMS - (System.currentTimeMillis() - _startTime);
+      long timeWaited = 0;
```
Comment (mayankshriv) [inline pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java]: Rename these to reflect time unit, ie something like connectionWaitTimeInMillis.

File: pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java [hunk 3]
```diff
@@ -426,11 +426,13 @@ private void runInternal() {
       KeyedFuture<ServerInstance, NettyClientConnection> keyedFuture = null;
       boolean gotConnection = false;
       boolean error = true;
+      long timeRemaining = _timeoutMS - (System.currentTimeMillis() - _startTime);
```
Comment (mayankshriv) [inline pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java]: Redundant initialization?

File: pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java [hunk 4]
```diff
@@ -32,6 +32,8 @@
   REQUEST_FETCH_EXCEPTIONS("exceptions", false),
   REQUEST_DESERIALIZATION_EXCEPTIONS("exceptions", false),
   DOCUMENTS_SCANNED("documents", false),
+  TIMEOUT_WAITING_FOR_CONNECTION("timeouts", false),
```
Comment (antumbde) [inline pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java]: connection_timeouts

File: pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java (hunk 1 above)
Comment (antumbde) [inline pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java]: connection_waits.

Something indicating what this wait is for

File: pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java (hunk 4 above)
Comment (rven1) [inline pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java]: Done

File: pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java (hunk 1 above)
Comment (rven1) [inline pinot-common/src/main/java/com/linkedin/pinot/common/metrics/BrokerMeter.java]: Done

File: pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java (hunk 3 above)
Comment (rven1) [inline pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java]: Removed the other initialization.

File: pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java (hunk 2 above)
Comment (rven1) [inline pinot-transport/src/main/java/com/linkedin/pinot/transport/scattergather/ScatterGatherImpl.java]: Done

**Pull request merged**